src/components/Footer.tsx

```diff
@@ -0,0 +1,62 @@
+import { useCallback, useContext } from 'react';
+import cn from 'classnames';
+import { GlobalContext } from '../context/GlobalContext';
+import { FilterType } from '../types/FilterTypes';
+
+type Props = {
+  status: string;
+  setStatus: (arg: FilterType) => void;
+};
+
+export const Footer: React.FC<Props> = ({ status, setStatus }) => {
+  const { todos, removeAllCompleted } = useContext(GlobalContext);
+
+  const includesCompletedTodo = useCallback(() => {
+    return todos.some((todo) => todo.completed);
+  }, [todos]);
+
+  const onClearCompletedButton = () => {
+    return removeAllCompleted();
+  };
+
+  return (
+    <footer className="todoapp__footer">
+      <span className="todo-count">{`${todos.length} items left`}</span>
+
+      <nav className="filter">
+        <a
+          href="#/"
+          className={cn('filter__link', { selected: status === 'all' })}
+          onClick={() => setStatus(FilterType.ALL)}
```

Comment:
suggest to make one handler factory for all filter button and call with proper value
```
const handleFilter = (filterType: FilterType) => () => setStatus(filterType);
```
```suggestion
          onClick={handleFilter(FilterType.ALL)}
```